### PR TITLE
fix(api): handle trailing / in MS transcript urls

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref: https://github.com/freeCodeCamp/freeCodeCamp/pull/51764 

I didn't recreate the tests since they're there for the new api.

### CODE REVIEW

**Review by naomi-lgbt — APPROVED** [review]

File: api-server/src/server/boot/user.js [hunk 1]
```diff
@@ -129,7 +140,7 @@ function createPostMsUsername(app) {
       });
     }
 
-    const msTranscriptId = msTranscriptUrl.split('/').pop();
+    const msTranscriptId = getMsTranscriptApiUrl(msTranscriptUrl);
     const msTranscriptApiUrl = `https://learn.microsoft.com/api/profiles/transcript/share/${msTranscriptId}`;
```
Comment (moT01) [inline api-server/src/server/boot/user.js:144]:
```suggestion
    const msTranscriptApiUrl = getMsTranscriptApiUrl(msTranscriptUrl);
```

The new function is returning the whole URL, not the transcript ID. So we were making requests to `https://learn.microsoft.com/api/profiles/transcript/share/https://learn.microsoft.com/api/profiles/transcript/share/abcdefghijklmno` (not correct)

**Review by moT01 — CHANGES_REQUESTED** [review]
Minor bug

File: api-server/src/server/boot/user.js (hunk 1 above)
Comment (ojeytonwilliams) [reply api-server/src/server/boot/user.js:144]: I should have written tests :face_exhaling:

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**